src/cuga/backend/tools/pptx_to_images.py

```
@@ -0,0 +1,611 @@
+
+
+def _ensure_shim() -> Path | None:
+    if _SHIM_SO.exists():
+        return _SHIM_SO
+    gcc = shutil.which("gcc")
+    if not gcc:
+        return None
+    src = Path(tempfile.gettempdir()) / "lo_socket_shim.c"
+    src.write_text(_SHIM_C)
+    try:
+        result = subprocess.run(
```

Ruff (0.15.20)

[error] `subprocess` call: check for execution of untrusted input

(S603)